sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/PrivateEndpoint.java

```diff
@@ -0,0 +1,263 @@
+        String privateLinkResourceId();
+
+        /**
+         * @return the collection of sub resource names.
+         */
+        List<PrivateLinkSubResourceName> subResourceNames();
+
+        /**
+         * @return the request message.
+         */
+        String requestMessage();
+
+        /**
+         * @return the provisioning state.
+         */
+        ProvisioningState provisioningState();
+
+        /**
+         * @return whether connection is manual approval.
+         */
+        boolean isManualApproval();
+
+        /** Grouping of private endpoint connection definition stages. */
+        interface DefinitionStages {
+            /**
+             * The first stage of the definition.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface Blank<ParentT>
+                extends PrivateLinkServiceConnection.DefinitionStages.WithPrivateLinkServiceResource<ParentT> {
+            }
+
+            /**
+             * The stage of the definition allowing to specify the resource of the private link service.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface WithPrivateLinkServiceResource<ParentT> {
+                /**
+                 * Specifies the resource of the private link service.
+                 *
+                 * @param privateLinkServiceResource the resource of the private link service
+                 * @return the next stage of the definition
+                 */
+                PrivateLinkServiceConnection.DefinitionStages.WithSubResource<ParentT> withResource(
+                    Resource privateLinkServiceResource);
+
+                /**
+                 * Specifies the resource of the private link service.
+                 *
+                 * @param privateLinkServiceResourceId the resource ID of the private link service
+                 * @return the next stage of the definition
+                 */
+                PrivateLinkServiceConnection.DefinitionStages.WithSubResource<ParentT> withResource(
+                    String privateLinkServiceResourceId);
+            }
+
+            /**
+             * The stage of the definition allowing to specify the sub resource that this private endpoint
+             * should connect to.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface WithSubResource<ParentT> {
+                /**
+                 * Specifies the sub resource.
+                 *
+                 * @param subResourceName the name of the sub resource
+                 * @return the next stage of the definition
+                 */
+                PrivateLinkServiceConnection.DefinitionStages.WithApprovalMethod<ParentT> withSubResource(
```

Comment:
For now, it fails for 2 (azure CLI deprecated `group-ids` parameter, and changed to `group-id` parameter).

For private link service, it might be empty.